front/lib/api/assistant/permissions.ts

```diff
@@ -0,0 +1,84 @@
+import type {
+  DustAppRunConfigurationType,
+  ModelId,
+  PostOrPatchAgentConfigurationRequestBody,
+} from "@dust-tt/types";
+import { isDustAppRunConfiguration, removeNulls } from "@dust-tt/types";
+import { uniq } from "lodash";
+import { Op } from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import { AppResource } from "@app/lib/resources/app_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import type { GroupResource } from "@app/lib/resources/group_resource";
+
+export async function listAgentConfigurationsForGroups(
+  auth: Authenticator,
+  groups: GroupResource[]
+) {
+  return AgentConfiguration.findAll({
+    attributes: ["sId", "groupIds"],
+    where: {
+      workspaceId: auth.getNonNullableWorkspace().id,
+      status: "active",
+      groupIds: {
+        [Op.overlap]: groups.map((g) => g.id),
+      },
+    },
+  });
+}
+
+export function getDataSourceViewIdsFromActions(
+  actions: PostOrPatchAgentConfigurationRequestBody["assistant"]["actions"]
+): string[] {
+  const relevantActions = actions.filter(
+    (action) =>
+      action.type === "retrieval_configuration" ||
+      action.type === "process_configuration" ||
+      action.type === "tables_query_configuration"
+  );
+
+  return removeNulls(
+    relevantActions.flatMap((action) => {
+      if (
+        action.type === "retrieval_configuration" ||
+        action.type === "process_configuration"
+      ) {
+        return action.dataSources.map(
+          (dataSource) => dataSource.dataSourceViewId
+        );
+      } else if (action.type === "tables_query_configuration") {
+        return action.tables.map((table) => table.dataSourceViewId);
+      }
+      return [];
+    })
+  );
+}
+
+export async function getAgentConfigurationGroupIdsFromActions(
+  auth: Authenticator,
+  actions: PostOrPatchAgentConfigurationRequestBody["assistant"]["actions"]
+): Promise<number[]> {
```

Comment:
So this function didn't change the only thing we change here is the process of updating right?

Reply:
:+1: